Silicon/NXP/iMX6Pkg/AcpiTables/Dsdt-PCIe.inc

```diff
@@ -0,0 +1,483 @@
+           // ACPI spec section 19.5.141
+           //
+
+           WordBusNumber (
+               ResourceProducer, // Specify bus ranged is passed to child devices
+               MinFixed,         // Specify min address is fixed
+               MaxFixed,         // Specify max address is fixed
+               PosDecode,        // Positive decode of bus number
+               0,                // AddressGranularity 2 power of 0
+               0,                // AddressMinimum - Minimum Bus Number
+               1,                // AddressMaximum - Maximum Bus Number
+               0,                // AddressTranslation - Set to 0
+               2)                // RangeLength - Number of Busses
+
+
+           //
+           // PCI memory space
+           //
+           Memory32Fixed(ReadWrite, 0x01100000, 0x00E00000, )
+       })
+
+       Return(RBUF)
+   }
+
+   //
+   // Declare the PCI Routing Table.
+   // This defines SPI mappings of the four line-based interrupts
+   // associated with the root complex and hierarchy below it.
+   // ACPI spec section 6.2.12
+   //
+   Name(_PRT, Package()
+   {
+
+       //
+       // Routing for device 0, all functions.
+       // Note: ARM doesn't support LNK nodes, so the third param
+       // is 0 and the fourth param is the SPI number of the interrupt
+       // line.
+       //
+
+       Package() {0x0000FFFF, 0, 0, 155}, // INTA
+       Package() {0x0000FFFF, 1, 0, 154}, // INTB
+       Package() {0x0000FFFF, 2, 0, 153}, // INTC
+       Package() {0x0000FFFF, 3, 0, 152}, // INTD/MSI
+   })
+
+   //
+   // Declare an _OSC (OS Control Hand-off) method which takes 4 arguments.
+   //
+   // Arguments:
+   //   Arg0  A Buffer containing a UUID
+   //   Arg1  An Integer containing a Revision ID of the buffer format
+   //   Arg2  An Integer containing a count of entries in Arg3
+   //   Arg3  A Buffer containing a list of DWORD capabilities
+   // Return Value:
+   //   A Buffer containing a list of capabilities
+   // ACPI spec Section 6.2.10,
+   // PCI FW spec, Section 4.5.
+   //
+
+   Name(SUPP,0) // PCI _OSC Support Field value
+   Name(CTRL,0) // PCI _OSC Control Field value
+   Method(_OSC, 4)
+   {
+
+       //
+       // Look for the PCI Host Bridge Interface UUID.
+       // Section 6.2.10.3
+       //
+
+       //
+       // Create DWord-addressable fields from the Capabilities Buffer
+       // Create CDW1 outside the test as it's used in the else clause.
+       // The first DWORD in the _OSC Capabilities Buffer contains bits
+       // that are generic to _OSC
+       // PCI Firmware Specification - 4.5.1.
+       //
+       // ACPI - 6.2.10
+       // CDW return value information
+       // Bit 0 - Reserved (not used)
+       // Bit 1 - _OSC failure. Platform Firmware was unable to process the request or query
+       // Bit 2 - Unrecognized UUID
+       // Bit 3 - Unrecognized Revision
+       // Bit 4 - Capabilities Masked
+       // All others � reserved

```

Comment:
Do you know what are these "?" characters? I see them in some places in the comments but don't know what are they.

Reply:
fixed